ably/realtime_client.go

```diff
@@ -71,12 +71,14 @@ func (c *Realtime) onReconnected(err *proto.ErrorInfo, isNewID bool) {
 	if err == nil /* RTN15c3 */ && !isNewID /* RTN15g3 */ {
 		return
 	}
-
 	for _, ch := range c.Channels.All() {
 		switch ch.State() {
 		// TODO: SUSPENDED
-		case StateChanAttaching, StateChanAttached:
+		case StateChanAttaching, //(RTN19b)
+			StateChanAttached:
 			ch.mayAttach(false, false)
+		case StateChanDetached: //(RTN19b)
```

Comment:
RTN19b: "If there are any pending channels i.e. in the ATTACHING or **DETACHING** state"

```suggestion
		case StateChanDetaching: //(RTN19b)
```